### ISSUE TITLE
Sync with Github Toggle won't turn off [Bug Title]

### ISSUE THREAD

**Issue by MatanForU** [amplication/amplication#1127]
**Describe the bug**
Sync with github toggle won't turn off

**Expected behavior**
when i click the active toggle it should turn it off

**Screenshots**
<img width="769" alt="Screen Shot 2021-01-14 at 0 50 20" src="https://user-images.githubusercontent.com/8940907/104519748-9f737780-5602-11eb-91d5-04aea4846c96.png">

**Desktop (please complete the following information):**
 - OS: Mojave 10.14.6
 - Browser: Chrome
 - Version: Version 87.0.4280.141

**Comment by hadialqattan** [comment]
Hi there, I'm gonna solve this issue 😊

Plz, assign me! 😇

**Comment by serv** [comment]
I am running into a similar problem when I run locally.

OS: Ubuntu 20.04.1 LTS
Browser: Brave
Version: v 1.19.86

Is the problem possibly caused by not giving Amplication app, Github API key or secret?

In the README, this requirement was not mentioned.

![amplication](https://user-images.githubusercontent.com/103456/105660110-26381800-5e7f-11eb-8162-3270001fb826.png)

**Comment by yuval-hazaz** [comment]
@serv you are correct.
You need a local env file with a key to your own GitHub app.
I will provide you with the details later today and will add a Relevant doc on our docs site and readme 

thanks

**Comment by yuval-hazaz** [comment]
Please follow these steps to connect your local server to GitHub

1. Go to https://github.com/settings/apps/new to create a new GitHub App
2. Give the app any name 
3. Set the Callback URL to 
```
http://localhost:3001
```

4. Copy and save the client secret and client id

5. Add a .env.local file in the root of the server directory 
```
packages\amplication-server\.env.local
```
6.  Add the following content to the file
```
GITHUB_SECRET_SECRET_NAME=use_local_settings
GITHUB_CLIENT_SECRET=**[client_secret_here]**
GITHUB_CLIENT_ID=**[client_id_here]**
GITHUB_SCOPE=user:email
GITHUB_REDIRECT_URI=http://localhost:3001/github/callback
GITHUB_APP_AUTH_SCOPE=user:email,repo,read:org
GITHUB_APP_AUTH_REDIRECT_URI=http://localhost:3001/github-auth-app/callback/{appId}
``
7. Replace **[client_secret_here]** with the client secret of your new app
8. Replace **[client_id_here]** with the client ID of your new app

9. Restart Amplication server

![image](https://user-images.githubusercontent.com/43705455/105817539-6b169880-5fbe-11eb-8bb3-94a9089080a8.png)

**Comment by serv** [comment]
Hi @yuval-hazaz ,

I was able to go through your steps ok. 

Now the problem is, the list of repositories are not updated with the latest list.

I first synced with Github successfully.

Then clicked on Select Repository, then I realized that I have to create a repository first. So I created a new repository. However, the new repo isn't showing up in the list even though I am clicking on the "refresh" icon.

**Comment by yuval-hazaz** [comment]
Thanks @serv, this should not happen. The list of repos is fetched from GitHub ordered by the "last updated" with the latest on top. 
In case you refreshed the entire page and the repo is still missing it must be something else. (did the repo was created for the same user who did the authentication in Amplication? )

If you want help with troubleshooting, please send me a message on discord and I will try to help
https://discord.gg/b8MrjU6

**Comment by yuval-hazaz** [comment]
resolved in #1197
